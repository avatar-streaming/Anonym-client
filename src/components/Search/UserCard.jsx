import React from "react";
import useCheckFollow from "../../hooks/useCheckFollow";
import useFollow from "../../hooks/useFollow";
import useUnfollow from "../../hooks/useUnfollow";

function UserCard({ _id, thumnail, userName, followers }) {
  const isFollow = useCheckFollow(_id);
  const followUser = useFollow();
  const unfollowUser = useUnfollow();

  return (
    <div className="user-card">
      <div className="user-card__left">
        <img
          className="user-card__thumnail user-thumnail"
          src={thumnail}
          alt="user thumnail"
        />
      </div>
      <div className="user-card__right">
        <div className="user-card__description">
          <div className="user-card__user-name">
            {userName}
          </div>
          <div className="follower-number">
            {followers.length} followers
          </div>
        </div>
        <div className="follow-button-box">
          {
            isFollow ? (
              <button
                className="follow-button"
                onClick={() => {
                  unfollowUser(_id);
                }}
              >
                unfollow
              </button>
            ) : (
              <button
                className="follow-button"
                onClick={() => {
                  followUser(_id);
                }}
              >
                follow
              </button>
            )
          }
        </div>
      </div>
    </div>
  );
}

export default UserCard;
