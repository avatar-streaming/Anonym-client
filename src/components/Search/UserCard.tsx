import useCheckFollow from "../../hooks/useCheckFollow";
import useFollow from "../../hooks/useFollow";
import useUnfollow from "../../hooks/useUnfollow";

<<<<<<< HEAD:src/components/Search/UserCard.tsx
interface UserCardProps {
  _id: string;
  thumnail: string;
  userName: string;
  followers: string[];
}

function UserCard({ _id, thumnail, userName, followers }: UserCardProps) {
=======
function UserCard({ _id, thumnail, userName, followers }) {
>>>>>>> c05b8641dad7cfa992c68cc1b3c69d422f4bf8ce:src/components/Search/UserCard.jsx
  const isFollow = useCheckFollow(_id);
  const followUser = useFollow();
  const unfollowUser = useUnfollow();

  return (
    <div className="user-card">
      <div className="user-card__left">
        <img
          className="user-card__thumnail user-thumnail"
          src={thumnail}
          alt="user thumnail"
        />
      </div>
      <div className="user-card__right">
        <div className="user-card__description">
          <div className="user-card__user-name">
            {userName}
          </div>
          <div className="follower-number">
            {followers.length} followers
          </div>
        </div>
        <div className="follow-button-box">
          {
            isFollow ? (
              <button
                className="follow-button"
                onClick={() => {
                  unfollowUser(_id);
                }}
              >
                unfollow
              </button>
            ) : (
              <button
                className="follow-button"
                onClick={() => {
                  followUser(_id);
                }}
              >
                follow
              </button>
            )
          }
        </div>
      </div>
    </div>
  );
}

export default UserCard;
