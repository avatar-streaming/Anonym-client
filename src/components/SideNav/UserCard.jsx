import React from "react";
import PropTypes from "prop-types";

function UserCard({ userInfo }) {
  return (
    <div className="following-card">
      <div>
        <img src={userInfo.thumnail} className="user-thumnail" alt="user thumnail" />
      </div>
      <div>
        <span className="user-name">{userInfo.userName}</span>
        <span className="streaming-state">Offline</span>
      </div>
    </div>
  );
}

export default UserCard;

UserCard.propTypes = {
  userInfo: PropTypes.shape({
    thumnail: PropTypes.string,
    userName: PropTypes.string,
  }),
};
