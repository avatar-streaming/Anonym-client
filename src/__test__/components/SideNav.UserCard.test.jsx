import React from "react";
import UserCard from "../../components/SideNav/UserCard";
import renderer from "react-test-renderer";

describe("<UserCard />", () => {
  let context = null;
  let component = null;

  beforeEach(() => {
    context = {
      thumnail: "mock thumnail",
      userName: "mock userName",
    };

    component = renderer.create(<UserCard user={context} />);
  });

  it("should render with given state from context", () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
