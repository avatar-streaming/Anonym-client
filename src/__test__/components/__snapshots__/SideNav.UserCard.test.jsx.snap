// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserCard /> should render with given state from context 1`] = `
<div
  className="following-card"
>
  <div>
    <img
      alt="user thumnail"
      className="following-card__thumnail user-thumnail"
      src="mock thumnail"
    />
  </div>
  <div>
    <span
      className="following-card__user-name"
    >
      mock userName
    </span>
    <span
      className="following-card__streaming-state"
    >
      Offline
    </span>
  </div>
</div>
`;
